### ISSUE TITLE
Pre-processing images

### ISSUE THREAD

**Issue by LancerXE** [rayguan97/GANav-offroad#15]
Hi, thanks for your patience! I have a question about the pre-processing part.
I have a small dataset which is Iabeled with 6 class (same with this project), but the processed images looked like this:
![left0290_orig](https://user-images.githubusercontent.com/98766249/181029156-2f19aaa3-58f3-4eb0-9ec7-57f0ff501d8c.png)
With the original ground truth:
![left0290](https://user-images.githubusercontent.com/98766249/181029245-f892fdfa-607d-4976-aec2-f3f0205421b9.png)
Here the background is white which is different from the processed images of the RUGD dataset, so I'm wondering if I made something wrong during the process.
Here is the code I changed from the 'rugd_relabel6.py' :

![code](https://user-images.githubusercontent.com/98766249/181033594-04252ba7-d543-4f47-b288-4e265a3809fa.PNG)

**Comment by rayguan97** [comment]
It’s very likely that the processing script is not suitable for your dataset. You need to figure out the logics for processing RUGD and apply to your dataset. I could not figure out the issue based on the information you provided. 

If the first img is the ID img, it should not have white color as the value is vary large. So it must be some bugs directly using rugd script for your dataset. You might have to go through your code and figure out the specific bugs. 

I’m closing the issue since it’s not about processing of rugd or rellis dataset.

**Comment by LancerXE** [comment]
Thanks for your reply!
Yes, it is the ID img, it seems that all the background are painted in white in that image.
I think I found the problem, in the '( )_relabel6.py' file , at line 35:
`color_id[tuple([0, 0, 0])] = 255`
This line of code will change the id of the class with RGB value (0, 0, 0) to 255, I'm not sure what is this line of code used for. 
When I run the original 'rugd_relabel6.py' in colab, I added two **print() function** to see the value of the parameter 'color_id', and it turns out after running the code `color_id[tuple([0, 0, 0])] = 255`, an extra value ` (0, 0, 0): 255` is added to the parameter 'color_id':
```
PALETTE = [ [ 108, 64, 20 ], [ 255, 229, 204 ],[ 0, 102, 0 ],[ 0, 255, 0 ],
            [ 0, 153, 153 ],[ 0, 128, 255 ],[ 0, 0, 255 ],[ 255, 255, 0 ],[ 255, 0, 127 ],
            [ 64, 64, 64 ],[ 255, 128, 0 ],[ 255, 0, 0 ],[ 153, 76, 0 ],[ 102, 102, 0 ],
            [ 102, 0, 0 ],[ 0, 255, 128 ],[ 204, 153, 255 ],[ 102, 0, 204 ],[ 255, 153, 204 ],
            [ 0, 102, 102 ],[ 153, 204, 255 ],[ 102, 255, 255 ],[ 101, 101, 11 ],[ 114, 85, 47 ] ]
```
```
color_id = {tuple(c):i for i, c in enumerate(PALETTE)}
print(color_id)
color_id[tuple([0, 0, 0])] = 255
print(color_id)
```
**result:**

`{(108, 64, 20): 0, (255, 229, 204): 1, (0, 102, 0): 2, (0, 255, 0): 3, (0, 153, 153): 4, (0, 128, 255): 5, (0, 0, 255): 6, (255, 255, 0): 7, (255, 0, 127): 8, (64, 64, 64): 9, (255, 128, 0): 10, (255, 0, 0): 11, (153, 76, 0): 12, (102, 102, 0): 13, (102, 0, 0): 14, (0, 255, 128): 15, (204, 153, 255): 16, (102, 0, 204): 17, (255, 153, 204): 18, (0, 102, 102): 19, (153, 204, 255): 20, (102, 255, 255): 21, (101, 101, 11): 22, (114, 85, 47): 23}`
`{(108, 64, 20): 0, (255, 229, 204): 1, (0, 102, 0): 2, (0, 255, 0): 3, (0, 153, 153): 4, (0, 128, 255): 5, (0, 0, 255): 6, (255, 255, 0): 7, (255, 0, 127): 8, (64, 64, 64): 9, (255, 128, 0): 10, (255, 0, 0): 11, (153, 76, 0): 12, (102, 102, 0): 13, (102, 0, 0): 14, (0, 255, 128): 15, (204, 153, 255): 16, (102, 0, 204): 17, (255, 153, 204): 18, (0, 102, 102): 19, (153, 204, 255): 20, (102, 255, 255): 21, (101, 101, 11): 22, (114, 85, 47): 23, (0, 0, 0): 255}`

So I tried to comment out this line of code for my own dataset, and after pre-processing the processed image looked like this:


![left0290_orig](https://user-images.githubusercontent.com/98766249/181222515-19a4cbcd-9326-407c-8dfa-49daf064bb9f.png)

There's no white color on the iD img, but I still feel that it doesn't looks right since it's all black.
